infoTree/createRecycleTree.js

```diff
@@ -0,0 +1,208 @@
+  }
+};
+Tree.prototype._postOrder = function(node, fn) {
+  if(node) {
+    for(var i = 0; i < node.children.length; i++) {
+      this._postOrder(node.children[i], fn);
+    }
+    if(fn) {
+      fn(node);
+    }
+  }
+};
+Tree.prototype.traverseDFS = function(fn, method) {
+  var current = this.root;
+  if(method) {
+    this['_' + method](current, fn);
+  } else {
+    this._preOrder(current, fn);
+  }
+};
+Tree.prototype.traverseBFS = function(fn) {
+  var queue = [this.root];
+  while(queue.length) {
+    var node = queue.shift();
+    if(fn) {
+      fn(node);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+};
+Tree.prototype.print = function() {
+  if(!this.root) {
+    return console.log('No root node found');
+  }
+  var newline = new Node('|');
+  var queue = [this.root, newline];
+  var string = '';
+  while(queue.length) {
+    var node = queue.shift();
+    string += node.data.toString() + ' ';
+    if(node === newline && queue.length) {
+      queue.push(newline);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+  console.log(string.slice(0, -2).trim());
+};
+Tree.prototype.printByLevel = function() {
+  if(!this.root) {
+    return console.log('No root node found');
+  }
+  var newline = new Node('\n');
+  var queue = [this.root, newline];
+  var string = '';
+  while(queue.length) {
+    var node = queue.shift();
+    string += node.data.toString() + (node.data !== '\n' ? ' ' : '');
+    if(node === newline && queue.length) {
+      queue.push(newline);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+  console.log(string.trim());
+};
+//TODO: takes in a leaf data string and returns it's address as a string formatted [num, num, ... num]
+//where tree.root.children[num].children[num]...children[num].data is the string put in
+Tree.prototype.findAddress = function(data){
+	if(!this.contains(data, this.traverseBF)){
+		return console.log("Tree does not contain "+ data);
+	}
+	return true
+}
+
+Tree.prototype.findEveryLeaf = function(){
+	var leafArray = []
+	//TODO: find every leaf Node and push it into leafArray
+	return leafArray;
+}
+
+//finds the URL to scrape data from by adding parent's name to base URL
+Tree.prototype.findUrl = function(leaf){
+	url = "https://www.seattle.gov/util/MyServices/WhereDoesItGo/"
+	while(leaf.data!=='rootNode'){
+		url += leaf.data + "/";
+		//Bad Idea. Will fix
+		leaf = leaf.parent;
+	}
+	url += "index.htm";
+	return url
+}
+// End Tree Implementation
+
+
+
+
+
+
+
+
+
+
+
+//formatting whereDoesItGo. Changes the value string => array. 
+//This could be done with regular expressions, but an array containing the parents seems easier to me.
+//TODO: adapt for different inputs. because of the slice(4,-1) this could break easily when we include the category titles
+function formatRecycleArray(whereRecycleArray){
```

Comment:
This uses the tree structure and is not part of it, so I would put it in a different file.